system/lib/wasmfs/wasmfs.cpp

```diff
@@ -56,12 +56,17 @@ long __syscall_dup(long fd) {
 
   return fileTable.add(openFile.unlocked());
 }
-
-__wasi_errno_t __wasi_fd_write(__wasi_fd_t fd,
-                               const __wasi_ciovec_t* iovs,
-                               size_t iovs_len,
-                               __wasi_size_t* nwritten) {
-  if (iovs_len < 0) {
+enum Position { overwrite, preserve };
+// Internal write function called by __wasi_fd_write and __wasi_fd_pwrite
+// Receives an open file state offset.
+// Optionally sets open file state offset.
+__wasi_errno_t writeAtOffset(__wasi_fd_t fd,
```

Comment:
```suggestion
static __wasi_errno_t writeAtOffset(__wasi_fd_t fd,
```
Static ensures it cannot be used in any other file. It makes it easier to read too, since it's a clear signal it is an internal detail of this file.